file.go

```diff
@@ -160,14 +160,15 @@ func (fi *File) Flush() error {
 		return err
 	}
 
-	defer fd.Close()
-
-	return fd.Flush()
+	// Close calls fd.Flush() and will do a full sync since it was opened with
+	// Sync=true. So, there is not need to call fd.Flush()
+	return fd.Close()
 }
 
 func (fi *File) Sync() error {
-	// just being able to take the writelock means the descriptor is synced
-	// TODO: Why?
+	// The descriptor lock is locked in File.Open and unlocked in
+	// fileDescriptor.Close. Taking the writelock ensures that this File is not
+	// open.
```

Comment:
I agree with the logic explained here but I think we should better document `desclock` instead. The vague 'coordination' should explicitly say: this lock makes sure there is either only one opened write-mode file descriptor or many read-mode ones.

Reply:
Updated comment to explain in more detail.

Reply:
I'd rather have this kind of explanation in the more central `desclock` definition than inside a function manipulation but it's good as it is so I consider this fixed.